src/inverted_index.c

```diff
@@ -1014,7 +1016,9 @@ int IR_SkipTo(void *ctx, t_docId docId, RSIndexResult **hit) {
     goto eof;
   }
 
-  if (IR_CURRENT_BLOCK(ir).lastId < docId || BufferReader_AtEnd(&ir->br)) {
+  const t_docId lastIdOfCurrentBlock = IR_CURRENT_BLOCK(ir).lastId;
+  if (lastIdOfCurrentBlock < docId || (BufferReader_AtEnd(&ir->br) && lastIdOfCurrentBlock != docId)) {
+    // We need to skip to a next block since the requested doc id isn't in the current block
```

Comment:
Why not just remove the `|| BufferReader_AtEnd(&ir->br)` from the condition? Is there something special we should do in that case?